fastlane/lib/util.rb

```diff
@@ -45,3 +45,86 @@ def travis?
 def circle?
   ENV['CIRCLECI'] == 'true'
 end
+
+# diff two hashes at a path and return the changed items
+def deps_diff(old_hash, new_hash, path)
+  (new_hash.dig(*path).to_a - old_hash.dig(*path).to_a).to_h
+end
+
+def npm_native_package_changed?
+  old_package = JSON.parse(sh "git show '#{source_branch}:package.json'")
+  new_package = JSON.parse(sh "git show '#{current_branch}:package.json'")
+
+  changed_dependencies = deps_diff(old_package, new_package, ['dependencies']).keys
+  changed_devdependencies = deps_diff(old_package, new_package, ['devDependencies']).keys
+
+  dep_name_regex = /react|jsc/
```

Comment:
Can we pull this out into a constant and freeze it?

À la `DEP_NAME_REGEXP = /react|jsc/.freeze` outside this method, and change your comparison to `dep =~ DEP_NAME_REGEXP`.